src/Microsoft.VisualStudio.Services.Agent/Util/ApiUtil.cs

```diff
@@ -14,7 +14,24 @@ public static class ApiUtil
         public static VssConnection CreateConnection(Uri serverUri, VssCredentials credentials)
         {
             VssClientHttpRequestSettings settings = VssClientHttpRequestSettings.Default.Clone();
-            settings.MaxRetryRequest = 5;
+
+            int maxRetryRequest;
+            if (!int.TryParse(Environment.GetEnvironmentVariable("agent.httpretry") ?? string.Empty, out maxRetryRequest))
```

Comment:
It would be nice to move these into const and follow the same convention as `VSTS_HTTP_PROXY`.